include/wasm.h

```diff
@@ -702,12 +700,78 @@ static inline void wasm_val_init_ptr(own wasm_val_t* out, void* p) {
 
 static inline void* wasm_val_ptr(const wasm_val_t* val) {
 #if UINTPTR_MAX == UINT32_MAX
+  assert(val->kind == WASM_I32);
   return (void*)(intptr_t)val->of.i32;
 #elif UINTPTR_MAX == UINT64_MAX
+  assert(val->kind == WASM_I64);
   return (void*)(intptr_t)val->of.i64;
 #endif
 }
 
+static inline void wasm_val_init_i32(own wasm_val_t* out, int32_t i) {
+  out->kind = WASM_I32;
+  out->of.i32 = i;
+}
+
+static inline int32_t wasm_val_i32(const wasm_val_t* val) {
+  assert(val->kind == WASM_I32);
+  return val->of.i32;
+}
+
+static inline void wasm_val_init_i64(own wasm_val_t* out, int64_t i) {
+  out->kind = WASM_I64;
+  out->of.i64 = i;
+}
+
+static inline int64_t wasm_val_i64(const wasm_val_t* val) {
+  assert(val->kind == WASM_I64);
+  return val->of.i64;
+}
+
+static inline void wasm_val_init_f32(own wasm_val_t* out, float32_t f) {
+  out->kind = WASM_F32;
+  memcpy(&out->of.i32, &f, sizeof(float32_t));
```

Comment:
Is the memcpy in these accessors actually useful, given that the float will be passed by value anyway?